rest_server/src/models/marketplace_item.js

```diff
@@ -7,25 +7,35 @@ class MarketplaceItem {
   constructor(sequelize, DataTypes) {
     this.orm = sequelize.define('MarketplaceItem', {
       id: {
-        type: DataTypes.UUID,
+        type: DataTypes.STRING,
         defaultValue: DataTypes.UUIDV4,
         primaryKey: true,
       },
       name: DataTypes.STRING,
       author: DataTypes.STRING,
       type: DataTypes.STRING,
+      dataType: DataTypes.STRING,
+      dataUrl: DataTypes.STRING,
       categories: DataTypes.ARRAY(DataTypes.STRING),
       tags: DataTypes.ARRAY(DataTypes.STRING),
       summary: DataTypes.STRING,
-      description: DataTypes.TEXT,
-      content: DataTypes.Object, // TODO: protocol validation in the future
-      submits: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 0 },
+      protocol: DataTypes.TEXT,
+      useNumber: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        defaultValue: 0,
+      },
       starNumber: {
         type: DataTypes.INTEGER,
         allowNull: false,
         defaultValue: 0,
       },
       status: DataTypes.ENUM('pending', 'approved', 'rejected'),
+      createdAt: {
```

Comment:
the createdAt value is set by rest server or database?

Reply:
rest server